app/smc/src/main.c

```diff
@@ -145,3 +147,25 @@ static int bh_arc_init_end(void)
 	return 0;
 }
 SYS_INIT_APP(bh_arc_init_end);
+
+#ifdef CONFIG_FPU_SHARING
+
+static void enable_float_on_current(struct k_work *work)
+{
+	k_float_enable(k_current_get(), 0);
+}
+
```

Comment:
Something like this in `main()` would probably work, or at least in an `app_init()` called from `main()`.
```suggestion
if (IS_ENABLED(CONFIG_FPU_SHARING)) {
	k_float_enable(k_current_get(), 0);
}
```

Generally, it's preferable to use `IS_ENABLED()` instead of `#ifdef` since the former ensures that code compiles even when an option is not enabled.

Reply:
Okay.